ev3dev/core.py

```diff
@@ -3300,3 +3300,254 @@ def get_volume(channel=None):
+            >>>     ('D5', 'h'),
+            >>>     ('C5', 'e3'),      # meas 2
+            >>>     ('B4', 'e3'),
+            >>>     ('A4', 'e3'),
+            >>>     ('G5', 'h'),
+            >>>     ('D5', 'q'),
+            >>>     ('C5', 'e3'),      # meas 3
+            >>>     ('B4', 'e3'),
+            >>>     ('A4', 'e3'),
+            >>>     ('G5', 'h'),
+            >>>     ('D5', 'q'),
+            >>>     ('C5', 'e3'),      # meas 4
+            >>>     ('B4', 'e3'),
+            >>>     ('C5', 'e3'),
+            >>>     ('A4', 'h.'),
+            >>> ))
+
+        .. important::
+
+            Only 4/4 signature songs are supported with respect to note durations.
+
+        Args:
+            song (iterable[tuple(str, str)]): the song
+            tempo (int): the song tempo, given in quarters per minute
+            delay (int): delay in ms between notes
+
+        Returns:
+            subprocess.Popen: the spawn subprocess
+        """
+        meas_duration = 60000 / tempo * 4
+
+        def beep_args(note, value):
+            """ Builds the arguments string for producing a beep matching
+            the requested note and value.
+
+            Args:
+                note (str): the note note and octave
+                value (str): the note value expression
+            Returns:
+                str: the arguments to be passed to the beep command
+            """
+            freq = NOTE_FREQUENCIES[note.upper()]
+            if '/' in value:
+                base, factor = value.split('/')
+                duration = meas_duration * NOTE_VALUES[base] / float(factor)
+            elif '*' in value:
+                base, factor = value.split('*')
+                duration = meas_duration * NOTE_VALUES[base] * float(factor)
+            elif value.endswith('.'):
+                base = value[:-1]
+                duration = meas_duration * NOTE_VALUES[base] * 1.5
+            elif value.endswith('3'):
+                base = value[:-1]
+                duration = meas_duration * NOTE_VALUES[base] * 2 / 3
+            else:
+                duration = meas_duration * NOTE_VALUES[value]
+
+            return '-f %d -l %d -D %d' % (freq, duration, delay)
+
+        return Sound.beep(' -n '.join(
+            [beep_args(note, value) for note, value in song]
+        ))
+
+
+def _make_scales(notes):
+    """ Utility function used for building the note frequencies table """
+    res = dict()
+    for note, freq in notes:
+        freq = round(freq)
+        for n in note.split('/'):
+            res[n] = freq
+    return res
+
+#: Note frequencies.
+#:
+#: This dictionary gives the rounded frequency of a note specified by its
+#: standard US abbreviation and its octave number (e.g. ``C3``).
+#: Alterations use the ``#`` and ``b`` symbols, respectively for
+#: *sharp* and *flat*, between the note code and the octave number (e.g. ``D#4``, ``Gb5``).
+NOTE_FREQUENCIES = _make_scales((
```

Comment:
Do this dictionary and the `_make_scales()` function above have use outside of the `Sound` class? I mean, would it make sense to move them into the class scope?